FirebaseAdmin/FirebaseAdmin/Auth/Hash/StandardScrypt.cs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2020, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+
+namespace FirebaseAdmin.Auth.Hash
+{
+    /// <summary>
+    /// Represents the Standard Scrypt password hashing algorithm. Can be used as an instance of
+    /// <a cref="UserImportHash">UserImportHash</a> when importing users.
+    /// </summary>
+    public sealed class StandardScrypt : UserImportHash
+    {
+        private int? derivedKeyLength;
+
+        private int? blockSize;
+
+        private int? memoryCost;
+
+        private int? parallelization;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StandardScrypt"/> class.
+        /// Defines the name of the hash to be equal to STANDARD_SCRYPT.
+        /// </summary>
+        public StandardScrypt()
+            : base("STANDARD_SCRYPT") { }
+
+        /// <summary>
+        /// Gets or sets the derived key length for the hashing algorithm.
+        /// <remarks>The length cannot be negative.</remarks>
+        /// </summary>
+        public int DerivedKeyLength
```

Comment:
Lets clean this all up by using regular properties and validating in `GetOptions`